### ISSUE TITLE
Tests are broken in release 1.64

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/hap#122]
Unfortunately the latest release 1.64 is not passing its own test suite in the GAP package distribution. It does pass the test suite here, though. Which made me wonder if there is a difference between the release archive and the code in this repository. Indeed that is the case:
```diff
diff -ru lib/ResolutionsModP/resPrimeGroup.gi tmp/hap-1.64/lib/ResolutionsModP/resPrimeGroup.gi
--- lib/ResolutionsModP/resPrimeGroup.gi	2024-07-29 11:42:04.000000000 +0200
+++ tmp/hap-1.64/lib/ResolutionsModP/resPrimeGroup.gi	2024-07-28 00:07:44.000000000 +0200
@@ -1,5 +1,5 @@
 #(C) Graham Ellis, 2005-2006
-
+RT:=0;
 #####################################################################
 #####################################################################
 InstallGlobalFunction(ResolutionPrimePowerGroup,
@@ -392,7 +392,7 @@
 #####################################################################
 SolutionMatBoundaryMatrices:=function(m,vec)
 local i,ncols,sem, vno, z,x, row, sol;
-
+RT:=RT-Runtime();
 ncols := Length(vec);
 z := zero;
 sol := ListWithIdenticalEntries(EchelonMatrices[m][2],z);
@@ -409,7 +409,7 @@
        fi;
     od;
        if IsZero(vec) then
-
+RT:=RT+Runtime();
       return sol;
         fi;
 
```

Another difference is that the release has a file `tst/testall3/bianchi.1.tst` that is not in the repository as far as I can tell.

So that definitely should be fixed. But I don't know if it alone explains [the many failing tests](https://github.com/gap-system/PackageDistro/actions/runs/10141434849/job/28039176424#step:8:667), such as this one:

```
########> Diff in /home/runner/gap/pkg/hap-1.64/tst/testextra2/3.10.1.tst:3
# Input is:
A:=ModPSteenrodAlgebra(G,8);;
# Expected output:
# But found:
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 1st choice method found for `-' on 2 arguments
########
########> Diff in /home/runner/gap/pkg/hap-1.64/tst/testextra2/3.10.1.tst:4
# Input is:
List(ModPRingGenerators(A),x->Bockstein(A,x));
# Expected output:
[ 0*v.1, 0*v.1, v.5, 0*v.1, (Z(3))*v.7+v.8+(Z(3))*v.9 ]
# But found:
Error, <obj> must be a component object or a record
########
```

**Comment by grahamknockillaree** [comment]
I spotted and corrected that difference and uploaded a new

lib/ResolutionsModP/resPrimeGroup.gi

file to github.  For some reason  I couldn't get the uploaded file to pass the github CI check.


Ah ha, but I see that @fingolfin has solved that problem.  So I'll upload a new version hap-1.65 later today.




Graham




________________________________
From: Max Horn ***@***.***>
Sent: Monday, July 29, 2024 11:03 AM
To: gap-packages/hap ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [gap-packages/hap] Tests are broken in release 1.64 (Issue #122)

EXTERNAL EMAIL: This email originated outside the University of Galway. Do not open attachments or click on links unless you believe the content is safe.
RÍOMHPHOST SEACHTRACH: Níor tháinig an ríomhphost seo ó Ollscoil na Gaillimhe. Ná hoscail ceangaltáin agus ná cliceáil ar naisc mura gcreideann tú go bhfuil an t-ábhar sábháilte.


Unfortunately the latest release 1.64 is not passing its own test suite in the GAP package distribution. It does pass the test suite here, though. Which made me wonder if there is a difference between the release archive and the code in this repository. Indeed that is the case:

diff -ru lib/ResolutionsModP/resPrimeGroup.gi tmp/hap-1.64/lib/ResolutionsModP/resPrimeGroup.gi
--- lib/ResolutionsModP/resPrimeGroup.gi        2024-07-29 11:42:04.000000000 +0200
+++ tmp/hap-1.64/lib/ResolutionsModP/resPrimeGroup.gi   2024-07-28 00:07:44.000000000 +0200
@@ -1,5 +1,5 @@
 #(C) Graham Ellis, 2005-2006
-
+RT:=0;
 #####################################################################
 #####################################################################
 InstallGlobalFunction(ResolutionPrimePowerGroup,
@@ -392,7 +392,7 @@
 #####################################################################
 SolutionMatBoundaryMatrices:=function(m,vec)
 local i,ncols,sem, vno, z,x, row, sol;
-
+RT:=RT-Runtime();
 ncols := Length(vec);
 z := zero;
 sol := ListWithIdenticalEntries(EchelonMatrices[m][2],z);
@@ -409,7 +409,7 @@
        fi;
     od;
        if IsZero(vec) then
-
+RT:=RT+Runtime();
       return sol;
         fi;


Another difference is that the release has a file tst/testall3/bianchi.1.tst that is not in the repository as far as I can tell.

So that definitely should be fixed. But I don't know if it alone explains the many failing tests<https://github.com/gap-system/PackageDistro/actions/runs/10141434849/job/28039176424#step:8:667>, such as this one:

########> Diff in /home/runner/gap/pkg/hap-1.64/tst/testextra2/3.10.1.tst:3
# Input is:
A:=ModPSteenrodAlgebra(G,8);;
# Expected output:
# But found:
Error, no method found! For debugging hints type ?Recovery from NoMethodFound
Error, no 1st choice method found for `-' on 2 arguments
########
########> Diff in /home/runner/gap/pkg/hap-1.64/tst/testextra2/3.10.1.tst:4
# Input is:
List(ModPRingGenerators(A),x->Bockstein(A,x));
# Expected output:
[ 0*v.1, 0*v.1, v.5, 0*v.1, (Z(3))*v.7+v.8+(Z(3))*v.9 ]
# But found:
Error, <obj> must be a component object or a record
########

**Comment by grahamknockillaree** [comment]
Fixed in version 1.65. Thanks for your help in getting the tests to run.
